### ISSUE TITLE
Requires inheriting from React.Component?

### ISSUE THREAD

**Issue by developit** [thejameskyle/babel-react-optimize#8]
Looking through the source for `babel-helpers-is-react-class`, it seems like the following two cases are not supported:

```js
import { Component } from 'react';
class Foo extends Component {
  // ...
}
```

```js
class Foo {
  // if you don't use setState, you don't need to inherit
}
```

The second case seems really desirable for transpiling to a pure function, since it's guaranteed to be a pure component (no setState).

Are these unsupported, or have I just missed some context? Cheers.

**Comment by agonbina** [comment]
Indeed, I also added the preset in a codebase where we extend from `Component`, which is imported as above, and it seems like none of the plugins are being applied. Any ideas?

**Comment by thejameskyle** [comment]
I can add support for:

```js
import { Component } from 'react';
class Foo extends Component {
  // ...
}
```

but not:

```js
class Foo {
  // if you don't use setState, you don't need to inherit
}
```

see this comment: https://github.com/thejameskyle/babel-react-optimize/issues/15#issuecomment-216949769

**Comment by developit** [comment]
@thejameskyle That's definitely a big step forward.  I don't find myself skipping inheriting from Component much anyway, since I believe that breaks `this.props`.

**Comment by akaila** [comment]
I am just curious what error do you get when you try and transpile a pure class (not deriving from React.Component). I get this:

Cannot read property 'name' of null

Can you please confirm. If so I cannot use this preset which is quite sad :(

**Comment by developit** [comment]
@akaila React itself does not support rendering components that don't inherit from `React.Component`. I think most of the other react-like frameworks do, certainly preact does. I don't think your error has anything to do with this preset.
